app/bundles/CoreBundle/Helper/CoreParametersHelper.php

```diff
@@ -58,14 +58,6 @@ public function all(): array
         return $this->resolvedParameters;
     }
 
-    /**
-     * @deprecated 3.0.0 to be removed in 4.0; use get() instead
-     */
-    public function getParameter($name, $default = null)
```

Comment:
I really like this, but it's a BC break. Either revert this method and we can merge the rest or this will have to wait for Mautic 6.

Reply:
I mean, I have no issues with this being merged into 6.x, but this is deprecated since 3.x **and** 4.x, although it was meant to be removed in 4.0 already.

Why wait to remove this if it shouldn't be in the code base after all?

On a side note: Is there a 6.x branch already?

Reply:
That's the whole point of major releases. That we can make these breaking changes. The fact that no one got to it for M3, M4 nor for M5 doesn't mean we can delete public methods in minor releases. Check [semver](https://semver.org/).

The method doesn't cause any troubles there. Only that it's ugly to have it there but otherwise there is no good reason to risk breaking some plugin with a minor Mautic release.

There is no branch for M6 just yet because there was no one who would take the responsibility of updating it with every Mautic 5 release.